File: iterativechop.py

def chop(element, lst):
    start = 0
    stop = len(lst) - 1
    while start <= stop:
        middle_idx = int((start + stop)/2)
        middle_element = lst[middle_idx]
        if middle_element == element:
            return middle_idx
        if middle_element < element:
            start = middle_idx + 1
        else:
            stop = middle_idx - 1
    return -1

File: chopr/iterativechop.rb
module IterChop
  def chop(item, list)
    start = 0
    stop = list.size - 1
    while start <= stop
      middle = ((start + stop) / 2).to_int
      middle_item = list[middle]
      if middle_item == item
        return middle
      end
      if middle_item < item
        start = middle + 1
      else
        stop = middle - 1
      end
    end
    return -1
  end
end
